src/components/text-input.components.js

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable default-param-last */
+/* eslint-disable unicorn/better-regex */
+import $ from 'jquery';
+
+export default class TextInput {
+    constructor(stringQuery, textEnable = false, required = true, textValid = '') {
+        this.stringQuery = stringQuery;
+
+        this.textInput = $(`${stringQuery} input[type="text"]`);
+
+        this.textValid = textValid;
+        this.textEnable = textEnable;
+        this.required = required;
+    }
+
+    getTextInput() {
+        return this.textInput.val();
+    }
+
+    setTextInput(value) {
+        this.textInput = this.textInput.val(value);
+    }
+
+    getTextValid() {
+        return this.textValid.val();
+    }
+
+    setTextValid(text) {
+        this.textValid = this.textInput.val(text);
+    }
+
+    isTextValid() {
+        if (this.textInput.val() === '') {
+            throw new Error('write any thing');
+        }
+
+        return this.textInput.val();
+    }
+
+    isTextEnable() {
```

Comment:
the state is wrong
![image](https://user-images.githubusercontent.com/23269934/162942665-83a4d9fe-9e82-45d1-805e-82d08b178a9d.png)

Reply:
How disabled input?

Reply:
Done!